pennylane/ops/cv.py

```diff
@@ -803,6 +807,63 @@ def _heisenberg_rep(p):
         return np.diag([-0.5, 0.5 / hbar, 0.5 / hbar])
 
 
+class TensorN(CVObservable):
+    r"""pennylane.ops.TensorN(wires)
+    The tensor product of the :class:`~.NumberOperator` acting on different wires.
+
+    If a single wire is defined, returns a :class:`~.NumberOperator` instance for convenient gradient computations.
+
+    When used with the :func:`~.expval` function, the expectation value
+    :math:`\langle \hat{n}_{i_0} \hat{n}_{i_1}\dots \hat{n}_{i_{N-1}}\rangle`
+    for a (sub)set of modes :math:`[i_0, i_1, \dots, i_{N-1}]` of the system is
+    returned.
+
+    **Example:**
+
+    >>> cv_obs = qml.TensorN(wires=[0, 1])
+    >>> cv_obs
+    TensorN(wires=[0, 1])
+    >>> cv_obs.ev_order is None
+    True
+
+    >>> cv_obs = qml.TensorN(wires=[1])
+    >>> cv_obs
+    NumberOperator(wires=[1])
+    >>> cv_obs.ev_order
+    2
```

Comment:
Nice! I think this is the first example for any observable/operation 😆

Reply:
Just double checking, does the following work correctly?

```python
cv_obs = qml.TensorN(wires=[0, 1])

@qml.qnode(dev)
def circuit():
    return qml.expval(cv_obs)
```

I think it should, whereas it wouldn't work for operations (since operations are queued on init).

Reply:
No error is raised with it

Reply:
Perhaps maybe we should hide this behind a `.. UsageDetails::` dropdown? Since it is nice to have, but I worry that having an example using a non-standard pattern will cause users to try and use operations as instances. Apart from that, this looks good to merge from my end!